packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/sdk.md

```diff
@@ -0,0 +1,111 @@
+> **Note**: The examples in this guide use Java 11 and Spring Boot MVC
+
+### 1: Start password recovery
+
+After the user starts the password recovery flow and selects the email authenticator for the process, Okta sends them an email matching the **Forgot Password** template you altered earlier.
+
+<div class="common-image-format">
+
+![Example of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png "Password recovery email")
+
+</div>
+
+When the user clicks the **Reset Password** link, their browser sends a request to the endpoint defined by the template, attaching the `${oneTimePassword}` and `${request.relayState}` VTL variables as query parameters to the URL. For instance, in the sample this request might render as `http://localhost:8080/magic-link/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`.
+
+### 2: Handle the OTP and state parameters
+
+Create a callback handler that takes the `${oneTimePassword}` and `${request.relayState}` values from the query string, saves them into local variables (for instance, 'otp', and 'state'), and makes the following checks:
+
+1. That `otp` and `state` are not `null`.
+2. That `state` matches the state value stored in the current `ProceedContext` session variable.
+
+If either check returns false, throw an error.
+
+```java
+@RequestMapping(value = {"magic-link/callback"}, method = RequestMethod.GET)
+public ModelAndView displayIndexOrHomePage(
+    final @RequestParam(name = "state", required = false) String state,
+    final @RequestParam(name = "otp", required = false) String otp,
+    final HttpSession session) 
+{
+    ProceedContext proceedContext = 
+        (ProceedContext)session.getAttribute("proceedContext");
+
+    // correlate received state with the client context
+    if (Strings.hasText(otp) && proceedContext != null &&
+        (Strings.isEmpty(state) || 
+         !state.equals(proceedContext.getClientContext().getState())))
+    {
+        ModelAndView mav = new ModelAndView("error");
+        mav.addObject("errors", 
+            "Could not correlate client context with the received state value " +
+            state + " in callback");
+        return mav;
+    }
+```
+
+If `ProceedContext` is `null` (because the user clicked the magic link from a different browser), advise the user to return to the original tab in the browser, where they requested the password reset, and enter the OTP to proceed.
```

Comment:
```suggestion
If `ProceedContext` is `null` (because the user clicked the magic link from a different browser), advise the user to return to the original tab in the browser where they requested the password reset, and then enter the OTP to proceed.
```

Reply:
fixed